lib/logflare/source/bigquery/pipeline.ex

```diff
@@ -206,4 +211,31 @@ defmodule Logflare.Source.BigQuery.Pipeline do
         )
     end
   end
+
+  # https://hexdocs.pm/broadway/Broadway.html#start_link/2
+  # split batch sizes based on json size
+  # ensure that we are well below the 10MB limit
+  defp bq_batch_size_splitter() do
+    {
+      {@max_batch_size, @max_batch_length},
+      fn
+        # reach max count, emit
+        _message, {1, _len} ->
+          {:emit, {@max_batch_size, @max_batch_length}}
+
+        # check content length
+        message, {count, len} ->
+          payload = Jason.encode!(message.data.body)
```

Comment:
Would be great if we could take the `payload` from here and use it in our BigQuery request so we don't have to encode twice.

Reply:
the google api sdk doesn't take in raw json string, so don't think it is possible  
https://github.com/Logflare/logflare/commit/bdea277e759abea845cbdae32996febcde1423fd